OmgvaPOS/src/DiscountManagement/Service/DiscountService.cs

```diff
@@ -0,0 +1,155 @@
+﻿using Microsoft.EntityFrameworkCore;
+using OmgvaPOS.Database.Context;
+using OmgvaPOS.DiscountManagement.Enums;
+using OmgvaPOS.DiscountManagement.Models;
+using OmgvaPOS.src.DiscountManagement.DTOs;
+using OmgvaPOS.src.DiscountManagement.Mappers;
+using OmgvaPOS.src.DiscountManagement.Repository;
+using OmgvaPOS.TaxManagement.Repository;
+using OmgvaPOS.HelperUtils;
+using OmgvaPOS.BusinessManagement.Models;
+using OmgvaPOS.ItemManagement.Repositories;
+using OmgvaPOS.src.ItemManagement.Mappers;
+using OmgvaPOS.ItemManagement.Services;
+using OmgvaPOS.ItemManagement.Models;
+
+namespace OmgvaPOS.src.DiscountManagement.Service;
+
+public class DiscountService : IDiscountService
+{
+    private readonly IDiscountRepository _discountRepository;
+    private readonly IItemRepository _itemRepository;
+    private readonly IItemService _itemService;
+    private readonly ILogger<DiscountService> _logger;
+    public DiscountService(
+        IDiscountRepository discountRepository, 
+        IItemRepository itemRepository,
+        IItemService itemService,
+        ILogger<DiscountService> logger
+        ) {
+        _discountRepository = discountRepository;
+        _itemRepository = itemRepository;
+        _itemService = itemService;
+        _logger = logger;
+    }
+
+    public Discount CreateDiscount(CreateDiscountRequest request) {
+        try {
+            DiscountValidator.ValidateDateCreate(request.TimeValidUntil);
+            DiscountValidator.ValidateDiscountAmount(request.Amount);
+            DiscountValidator.ValidateDiscountType(request);
+        }
+        catch (Exception ex) {
+            _logger.LogError(ex, "Error while validating create discount request.");
+            throw;
+        }
+
+        var discount = DiscountMapper.FromCreateDiscountRequest(request);
+        _discountRepository.AddDiscount(discount);
+        
+        //TODO:
+        if (request.Type == DiscountType.Order) {
+            throw new NotImplementedException("Order is not implemented yet");
+            //var order = _orderService.GetOrderById(request.OrderId);
+            //order.DiscountId = discount.Id;
+            //_orderService.UpdateOrder(order);
+        }
+
```

Comment:
Discount may also be applied to order item. I am not sure if it is accounted for.